pdns/pdns_recursor.cc

```diff
@@ -918,6 +920,54 @@ static void protobufLogResponse(pdns::ProtoZero::RecMessage& message)
   }
 }
 
+static void protobufLogResponse(const struct dnsheader* dh,
+                                const RecursorPacketCache::OptPBData& pbData, const struct timeval& tv,
+                                bool tcp, const ComboAddress& source, const ComboAddress& destination,
+                                const EDNSSubnetOpts& ednssubnet,
+                                const boost::uuids::uuid& uniqueId, const string& requestorId, const string& deviceId,
+                                const string& deviceName) 
+{
+  pdns::ProtoZero::RecMessage pbMessage(pbData ? pbData->d_message : "", pbData ? pbData->d_response : "", 64, 10); // The extra bytes we are going to add
+  if (pbData) {
+    // We take the inmutable string from the cache and are appending a few values
```

Comment:
empty?

Reply:
This is a kind of "cannot happen situation". There is a hit in the PC, but no protobuf packet form the cache. In that case we start with an empty one (instead of once having various things filled in) and add the bare minimum in the `else` branch of the `if (pbData)` block.

Reply:
so like this:

```cpp
if (!pbData) {
  // content of the current else block
}
```

Reply:
A, now I see what you mean. I was looking at the limited github view. Will fix.